### PR TITLE
migrate odp | Api Endpoint: /api/assessment/:name/originalDataPoint/:cycleName/:odpId

### PR DESCRIPTION
**Pull request by sorja**
image

```
curl localhost:9001/api/assessment/fra/originalDataPoint/2025/72 | jq 'keys'
  
[
  "countryIso",
  "dataSourceAdditionalComments",
  "dataSourceMethods",
  "dataSourceReferences",
  "description",
  "id",
  "idLegacy",
  "nationalClasses",
  "props",
  "year"
]
```

Todo
- [x] Api endpint
- [x] Add odp type to @meta/

Part of #1105

### CODE REVIEW

File: common/api/endpoint/ApiEndPoint.ts [hunk 1]
```diff
@@ -7,6 +7,10 @@ export const ApiEndPoint = {
   },
   Assessment: {
     sections: (name = ':name', cycleName = ':cycleName') => apiPath('assessment', name, 'sections', cycleName),
+    OriginalDataPoint: {
+      one: (name = ':name', cycleName = ':cycleName', odpId = ':odpId') =>
+        apiPath('assessment', name, 'originalDataPoint', cycleName, odpId),
```
Comment (minotogna) [inline common/api/endpoint/ApiEndPoint.ts:12]: path should be /assessments/:assessmentName/cycles/:cycleName/originalDataPoints/:originalDataPointId
that is the convention. should we change all the api paths in a separate pr ?

File: tools/dataMigration/migrateData/migrateOdps.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { DBNames } from '../_DBNames'
+
+export const migrateOdps = async (props: { assessment: Assessment }, client: ITask<any>): Promise<void> => {
+  const { assessment } = props
+  const pgp = pgPromise()
+
+  const queries = assessment.cycles.map((cycle) => {
+    const cycleName = DBNames.getCycleSchema(assessment.props.name, cycle.name)
+    return `
+    with c as (select jsonb_build_object('cycles', jsonb_agg(uuid)) as props from public.assessment_cycle)
```
Comment (minotogna) [inline tools/dataMigration/migrateData/migrateOdps.ts:14]: we dont need to add cycles to odp props. it is already in a cycle schema

Comment (minotogna) [reply tools/dataMigration/migrateData/migrateOdps.ts:14]: I think we dont need props at all

File: common/api/endpoint/ApiEndPoint.ts (hunk 1 above)
Comment (sorja) [reply common/api/endpoint/ApiEndPoint.ts:12]: ok

**Review by minotogna — APPROVED** [review]

**Pull request merged**